### PR TITLE
i/6145: Replaced the getResizeObserver() helper with the ResizeObserver class

### PR DESCRIPTION
**Pull request by oleq**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other: Replaced the `getResizeObserver()` helper with the `ResizeObserver` class for performance reasons. Closes ckeditor/ckeditor5#6145.

MINOR BREAKING CHANGE: The `getResizeObserver()` helper is no longer available. We recommend using the [`ResizeObserver`](http://ckeditor.com/docs/ckeditor5/latest/api/module_utils_dom_resizeobserver-ResizeObserver.html) class instead.

---

### Additional information

Requires: 

* [ ] https://github.com/ckeditor/ckeditor5-ui/pull/538 that aligns `ToolbarView` to the new API.

### CODE REVIEW

File: src/dom/resizeobserver.js [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+/**
+ * A helper class which instances allow performing custom actions when native DOM elements are resized.
+ *
+ *		const editableElement = editor.editing.view.getDomRoot();
+ *
+ *		const observer = new ResizeObserver( editableElement, entry => {
+ *			console.log( 'The editable element has been resized in DOM.' );
+ *			console.log( entry.target ); // -> editableElement
+ *			console.log( entry.contentRect.width ); // -> e.g. 423px
+ *		} )
```
Comment (Reinmar) [inline src/dom/resizeobserver.js]:
```suggestion
 *		} );
```

File: src/dom/resizeobserver.js [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+		ResizeObserver._observerInstance.observe( element );
+	}
+
+	/**
+	 * Destroys the observer which disables the `callback` passed to the {@link #constructor}.
+	 */
+	destroy() {
+		ResizeObserver._deleteElementCallback( this._element, this._callback );
+
+		this._element = this._callback = null;
```
Comment (Reinmar) [inline src/dom/resizeobserver.js]: Why?

Comment (Reinmar) [reply src/dom/resizeobserver.js]: No, that's not necessary for memory management.

File: src/dom/resizeobserver.js [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+
+/**
+ * A helper class which instances allow performing custom actions when native DOM elements are resized.
+ *
+ *		const editableElement = editor.editing.view.getDomRoot();
+ *
+ *		const observer = new ResizeObserver( editableElement, entry => {
+ *			console.log( 'The editable element has been resized in DOM.' );
+ *			console.log( entry.target ); // -> editableElement
+ *			console.log( entry.contentRect.width ); // -> e.g. 423px
```
Comment (Reinmar) [inline src/dom/resizeobserver.js]:
```suggestion
 *			console.log( entry.contentRect.width ); // -> e.g. '423px'
```

**Comment by Reinmar** [comment]
OK, from the code perspective this change looks good. There are 3 minor issues that I spotted. I leave the rest to you, @panr. Please verify how does it affect the case that you have, guys. And if all's fine and you like the implementation, you can merge those two PRs.

**Comment by Reinmar** [comment]
PS. I updated the merge message. It's a minor BC to me so I wanted to make that explicite.

**Comment by panr** [comment]
LGMT 👌 Merging it.

**Pull request merged**